server/src/main/java/org/opensearch/search/aggregations/bucket/terms/GlobalOrdinalsStringTermsAggregator.java

```diff
@@ -136,16 +144,95 @@ public GlobalOrdinalsStringTermsAggregator(
                 return new DenseGlobalOrds();
             });
         }
+        this.fieldName = (valuesSource instanceof ValuesSource.Bytes.WithOrdinals.FieldData)
+            ? ((ValuesSource.Bytes.WithOrdinals.FieldData) valuesSource).getIndexFieldName()
+            : null;
     }
 
     String descriptCollectionStrategy() {
         return collectionStrategy.describe();
     }
 
+    public void setWeight(Weight weight) {
+        this.weight = weight;
+    }
+
+    /**
+     Collects term frequencies for a given field from a LeafReaderContext.
+     @param ctx The LeafReaderContext to collect terms from
+     @param globalOrds The SortedSetDocValues for the field's ordinals
+     @param ordCountConsumer A consumer to accept collected term frequencies
+     @return A no-operation LeafBucketCollector implementation, since collection is complete
+     @throws IOException If an I/O error occurs during reading
+     */
+    LeafBucketCollector termDocFreqCollector(
+        LeafReaderContext ctx,
+        SortedSetDocValues globalOrds,
+        BiConsumer<Long, Integer> ordCountConsumer
+    ) throws IOException {
+        if (weight == null || weight.count(ctx) != ctx.reader().maxDoc()) {
+            // weight.count(ctx) == ctx.reader().maxDoc() implies there are no deleted documents and
+            // top-level query matches all docs in the segment
+            return null;
+        }
+
+        if (weight.count(ctx) == 0) {
+            // No documents matches top level query on this segment, we can skip the segment
+            return LeafBucketCollector.NO_OP_COLLECTOR;
+        }
+
+        Terms segmentTerms = ctx.reader().terms(this.fieldName);
+        if (segmentTerms == null) {
+            // Field is not indexed.
+            return null;
+        }
+
+        NumericDocValues docCountValues = DocValues.getNumeric(ctx.reader(), DocCountFieldMapper.NAME);
+        if (docCountValues.nextDoc() != NO_MORE_DOCS) {
+            // This segment has at least one document with the _doc_count field.
+            return null;
+        }
+
+        TermsEnum indexTermsEnum = segmentTerms.iterator();
```

Comment:
Can you add some comments here on what we are trying to achieve?

Also, there are no tests for this block.  Would it be possible to add a dataset that exercises this part of the code base?

Reply:
I can add some comments here. 

Regarding coverage - Please refer to my comment [here](https://github.com/opensearch-project/OpenSearch/pull/11643#issuecomment-1953247310).

Reply:
Integration tests don't count towards the coverage numbers, only unit tests do.

I believe it should be possible to hit this logic from `KeywordTermsAggregatorTests`. I'm kind of surprised that it doesn't get hit by `testMatchAllDocs`. If you debug it, you should be able to figure out why you're not getting here. Maybe add more tests to that class.

Reply:
Thanks for pointing `KeywordTermsAggregatorTests `. Actually in `testMatchAllDocs`, `ctx.reader().terms(this.fieldName)` is returning null at the moment which I why those code blocks are getting skipped.

Need to check why, probably `ctx.reader().terms` is not assigned correctly in these tests.

Reply:
Found why the current segmentTerms was being set to null. 
In existing tests we were adding fields to documents with [SortedSetDocValuesField](https://lucene.apache.org/core/7_4_0/core/org/apache/lucene/document/SortedSetDocValuesField.html) assignment. 

As per the documentation: "If you also need to store the value, you should add a separate [StoredField](https://lucene.apache.org/core/7_4_0/core/org/apache/lucene/document/StoredField.html) instance."

So using the above StoredField instance, the segmentTerms are getting assigned as expected. Adding test cases now to cover both the scenarios.

Reply:
It's less about stored fields and more about indexed fields.

It might be easier to add the fields using `KeywordField`, since it's indexed and has doc values.

Reply:
Makes sense, renaming `stored values` to `indexed fields` in associated variable names.